pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -93,29 +117,46 @@ func (b *BuildPlanByProject) Build() (*Build, error) {
 		return nil, errs.New("Project is nil")
 	}
 
-	if b.Project.Error != nil && b.Project.Message != "" {
-		return nil, errs.New(b.Project.Message)
+	if b.Project.Error != nil {
+		if b.Project.Error.Message != "" {
+			return nil, errs.New(b.Project.Message)
```

Comment:
Please wrap so we can trace the origin.

Reply:
The `Error` here is just a struct with a `Message` field. There is no error (of the `error` type) to wrap.

Reply:
I think one of us is confused with what the other means, genuinely not sure which one :p What I meant is I would expect something like:

```
errs.New("API returned project error message: " + b.Project.Message)
```

And I'd expect them to each be unique for the flagged errors. The point being that when we see this error we can trace where it originated in the code. If we just give the error the API gave and don't wrap it then it may be impossible to trace where this error was triggered in the code.

Reply:
Ah okay, that makes sense. I thought you were asking to use `errs.Wrap()` which is where the confusion came from.

Reply:
Yeah it's a bit weird since you have an error, it's just not typed. And you want to wrap it. Technically you could certainly do an `errs.Wrap()` here but it feels a little pointless.